server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -96,6 +96,21 @@ public static PrioritizedEsThreadPoolExecutor newSinglePrioritizing(
         return new PrioritizedEsThreadPoolExecutor(name, 1, 1, 0L, TimeUnit.MILLISECONDS, threadFactory, contextHolder, timer);
     }
 
+    /**
+     * Creates a scaling {@link EsThreadPoolExecutor} using an unbounded work queue.
+     * <p>
+     * The {@link EsThreadPoolExecutor} scales the same way as a regular {@link ThreadPoolExecutor} until the core pool size
+     * (and at least 1) is reached: each time a task is submitted a new worker is added regardless if an idle worker is available.
```

Comment:
"each time a task is submitted a new worker is added regardless if an idle worker is available"

Is this related to EsThreadPoolExecutor, to the JDK ThreadPoolExecutor, or both? (maybe specify which of the 3 so it's clear)

Reply:
that applies to all,i'll add that 👍

Reply:
> The {@link EsThreadPoolExecutor} scales the same way as a regular {@link ThreadPoolExecutor} until the core pool size (and at least 1) is reached

@ldematte wondering, isn't this clear enough?

Reply:
Yes, that makes clear to me that this applies to all

Reply:
👍